illumos-utils/src/zfs.rs

```diff
@@ -377,8 +395,20 @@ impl Zfs {
                 });
             }
         }
+        if let Some(reservation) = reservation {
+            if let Err(err) =
+                Self::set_value(name, "reservation", &format!("{reservation}"))
+            {
+                return Err(EnsureFilesystemError {
+                    name: name.to_string(),
+                    mountpoint: mountpoint.clone(),
+                    // Take the execution error from the SetValueError
+                    err: err.err.into(),
+                });
+            }
+        }
         if let Some(compression) = compression {
```

Comment:
If this is `None`, and there exists a reservation already, what happens?

Reply:
Updated in c5621256ccc7a8995842d057e608c8b3b66effe5, but now providing `None` means that the reservation will be deleted.

(I prefer this option to the three-way "set / unset / maybe-ignore" API, because Nexus is going to be telling sleds *exactly* what configuration to use anyway, and that seems easier to manage than something which could possibly require read-modify-write loops)